bookkeeper-server/src/main/java/org/apache/bookkeeper/http/DecommissionService.java

```diff
@@ -74,20 +73,28 @@ public HttpServiceResponse handle(HttpServiceRequest request) throws Exception {
                     BookieSocketAddress bookieSrc = new BookieSocketAddress(
                       bookieSrcString[0], Integer.parseInt(bookieSrcString[1]));
 
-                    admin.decommissionBookie(bookieSrc);
-                    admin.close();
+                    Thread thread = new Thread() {
+                        @Override
+                        public void run() {
+                            try {
+                                bka.decommissionBookie(bookieSrc);
+                            } catch (Exception e) {
+                                LOG.error("Error handling decommissionBookie: {} with exception {}", bookieSrc, e);
+                            }
+                        }
+                    };
+
+                    thread.run();
```

Comment:
this is exactly the same as what was previously there. I think you meant thread.start().

Reply:
Thanks. will change it.